.github/workflows/update-trivy-cache.yml

```diff
@@ -0,0 +1,41 @@
+name: Update Trivy Cache
+
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Run daily at midnight UTC
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  update-trivy-db:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup oras
+        uses: oras-project/setup-oras@9c92598691bfef1424de2f8fae81941568f5889c # v1.2.1
+
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Prepare DB directory
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/.cache/trivy/db
+      
+      - name: Download vulnerability database
+        uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3.0.0
+        with:
+          max_attempts: 10
+          retry_on: error
+          timeout_seconds: 30
+          retry_wait_seconds: 60
+          command: |
+            oras pull ghcr.io/aquasecurity/trivy-db:2
+            tar -xzf db.tar.gz -C $GITHUB_WORKSPACE/.cache/trivy/db
+            rm db.tar.gz
+
+      - name: Cache DBs
+        uses: actions/cache/save@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
+        with:
+          path: ${{ github.workspace }}/.cache/trivy
```

Comment:
We have a 10gb limit on the caches and already have issues maintaining size if too many PRs are open :). Is there a pruning mechanism to clean up from the cache?

Reply:
That's good call. The estimate size of trivy db is 38 MB. And github would remove the cache over 7 days: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy
Therefore it would take ~280 MB for 7-day cache.
![image](https://github.com/user-attachments/assets/58f797f4-2a77-4b98-b4b3-40d3141b075e)
Comparing with other cache entries, the trivy DB is rather small.
![image](https://github.com/user-attachments/assets/c7f4eceb-5d2a-4dc0-9f6a-9eb558e5eed3)
And looking into Ratify's cache, it only keeps entries in the last 2 days and evicts older entries as it already exceeds 10 GB limit.
I would say it should be safe to add this trivy cache, but I have a concern on the cache eviction. If the latest trivy db is evicted, then related pipelines would fail. However, as each PR or push event would trigger those jobs to query the db cache, so I assume it should not be evicted by github since the eviction policy is removing least recently used caches.

Reply:
Got it makes sense thanks